### ISSUE TITLE
[$250] Next steps are incorrect when there is no approval on a report

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#42834]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.77-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @kevinksullivan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1717026958410769

## Action Performed:

1. Set up workspace with no approvals configured
2. assign expensify card to a member on that workspace
3. Incur a card transaction

## Expected Result:
The next steps read should show:

Header: "Finished!"
Subheader: "No further action required!"

## Actual Result:
Next steps say this report needs approval.

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

![image (4)](https://github.com/Expensify/App/assets/38435837/80bc13e1-10fd-4470-9589-949408f32e08)

![image (3)](https://github.com/Expensify/App/assets/38435837/70ada7a6-6954-4ad5-adc0-608135938566)

![Snip - (72) New Expensify - Google Chrome (2)](https://github.com/Expensify/App/assets/38435837/e240019c-9e9a-44c6-b55e-ae91a1e3b619)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017287c932cce9e705</li>
        <li>Upwork Job ID: 1797365106853683200</li>
        <li>Last Price Increase: 2024-08-08</li>
<li>Automatic offers: </li>
<ul>
<li>alitoshmatov | Reviewer | 103271393</li>
</ul></li>
    </ul>
</details>

**Comment by MitchExpensify** [comment]
What do you think the header should say  @kevinksullivan ? Something like this?;

"Waiting for approval."

**Comment by MitchExpensify** [comment]
Going to make a call and suggest we remove next steps entirely in this situation so I've updated the OP accordingly.

Waiting on proposals at this point 👀

**Comment by MitchExpensify** [comment]
What wave collect status do you think this one fits under @trjExpensify ?

**Comment by trjExpensify** [comment]
Hm, is this part of the pending/scanning implementation? @kevinksullivan didn't we talk about this one using the "finished" terminology for the closed non-reimbursable reports?

**Comment by MitchExpensify** [comment]
Bumping Slack thread here https://expensify.slack.com/archives/C049HHMV9SM/p1718225472265049?thread_ts=1717026958.410769&cid=C049HHMV9SM

**Comment by MitchExpensify** [comment]
Updated the expected behavior based on [the slack conversation](https://expensify.slack.com/archives/C049HHMV9SM/p1718299069269899?thread_ts=1717026958.410769&cid=C049HHMV9SM) 

We need proposal now

**Comment by MitchExpensify** [comment]
I feel like this might be hot pick worthy, what d'you reckon @trjExpensify ?

**Comment by alitoshmatov** [comment]
> assign expensify card to a member on that workspace

I think we are not going to have any proposals since it requires expensify card

**Comment by trjExpensify** [comment]
>I feel like this might be hot pick worthy, what d'you reckon @trjExpensify ?

It's external isn't it? @grgia any idea how someone can use dev to simulate what we're asking for here in the OP?

**Comment by grgia** [comment]
@trjExpensify I believe this is known cc @kevinksullivan 

but the solution is to not soley base next steps off of report state, since the report can be in processing state and not require next steps. For this case, we would also want to check if the report has only non reimbursable transactions

**Comment by trjExpensify** [comment]
>@trjExpensify I believe this is known cc @kevinksullivan

As in, there's another issue for it? This is an issue that was reported by Kevin, so maybe it's this?

**Comment by MitchExpensify** [comment]
Yeah I think this is the only issue for this after a search

**Comment by grgia** [comment]
I think this is an internal issue, we likely need to update the nextSteps for this report to be finished if the report contains only non-reimbursable expenses. Next steps were implemented in the [Drafts project](https://docs.google.com/document/d/1vZtGpW_Fy-Wq_R6gvnjdmWVA86FnR42y3LtnNA--vcY/edit?pli=1). cc @mountiny if you have any quick input on where we'd need to add that check

@kevinksullivan @trjExpensify what is the expected header for this? / Does it also depend on policy type?

**Comment by mountiny** [comment]
@grgia is this only optimistic issue? So when we load the next steps from server its correct? 

If its coming incorrect from the server this is already an OldDot issue that we would need to fix in Web (same next steps would show up in oldDot)

**Comment by trjExpensify** [comment]
Looking at the screenshot in the OP: 

![image (4)](https://github.com/Expensify/App/assets/38435837/80bc13e1-10fd-4470-9589-949408f32e08)

Both of those transactions on the report look like they're pending, so should the expected results really be to use the "Finished" next steps in this case? 

In the doc it looks like we have a specific next step message for that case, which I would assume we want to copy to the report level when all expenses on the report are pending transactions? 

<img width="269" alt="image" src="https://github.com/Expensify/App/assets/16232057/a28ca5aa-11a7-4900-b9dc-e0e4e2b2d123">

@kevinksullivan it might be an idea if you could just list out the next steps and the cases in this issue for the expected results.

**Comment by MitchExpensify** [comment]
Not overdue, Melvin - Actively being worked on.

**Comment by MitchExpensify** [comment]
Not overdue, Melvin - Actively being worked on.

**Comment by grgia** [comment]
>  I think we are not using the expense level next steps in Newdot. So you would have to add it to the report level to show it

@mountiny  could you expand on this?

**Comment by grgia** [comment]
bump on above @mountiny

**Comment by mountiny** [comment]
@grgia in OldDot, you can have 2 next steps messages. One is shown in the expense report level and another next step information can be shown in the detail of transaction/ expense.

There are not many of these expenseMessages; you can see one [here](https://github.com/Expensify/Web-Expensify/blob/0bfc615234e6f411b7ecd36e4450a86458653c70/lib/Report/NextStep.php#L383-L385), for example.

We only show the `nextStep.message` in Newdot, not the `expenseMessage`

I think to show this message in the transaction thread 
![image](https://github.com/Expensify/App/assets/36083550/22080004-3b26-4896-b517-1af8390d4ab0)
we should not use report next steps, but rather use logic that tells us that the expense is still pending to show this message.

**Comment by MitchExpensify** [comment]
Did that resolve your question @grgia ? Do we need anything else at this point to solicit proposals based on the OP?

**Comment by grgia** [comment]
@MitchExpensify nothing is blocking proposals as is

**Comment by grgia** [comment]
waiting for proposals melvin

**Comment by MitchExpensify** [comment]
Reassigning as I'm heading on leave imminently 🙇

**Comment by pasyukevich** [comment]
Hi, I'm Eugene from Callstack - expert contributor group.  I’d like to work on this issue.

**Comment by trjExpensify** [comment]
Assigned!

**Comment by grgia** [comment]
@pasyukevich thanks for taking this one! Let me know if you need anything

**Comment by pasyukevich** [comment]
@grgia Could you please provide more details on how to reproduce this issue? I was able to add a card to the workspace without needing approval and successfully submitted an expense. However, I do not see the next steps text here. Maybe I am missing something.

**Comment by grgia** [comment]
@pasyukevich can you show me a screenshot of the expense?

**Comment by anmurali** [comment]
@pasyukevich bump on https://github.com/Expensify/App/issues/42834#issuecomment-2265175723

**Comment by pasyukevich** [comment]
@grgia 

Sure, 

<img width="1911" alt="image" src="https://github.com/user-attachments/assets/875177bf-f11f-415c-b12e-b243ddb139d9">

<img width="1911" alt="image" src="https://github.com/user-attachments/assets/07c068df-3de5-4938-96c3-950a54bcce28">

**Comment by grgia** [comment]
I haven't had a chance to look at this yet

**Comment by grgia** [comment]
@pasyukevich it looks like this might be specific to ECard transactions, which need to be QAd internally. @kevinksullivan @anmurali could you confirm

**Comment by grgia** [comment]
Awaiting confirmation on expected result here

**Comment by grgia** [comment]
solution
if the report is a nonreimbursable-only report, then display:

Header: "Finished!"
Subheader: "No further action required!"

**Comment by grgia** [comment]
Chatted [here](https://expensify.slack.com/archives/C036QM0SLJK/p1724271030141389)

There's three cases to fix:

if single expense report and all expenses are pending or scanning show the pending/scanning banner
if single expense report and all expenses are finished show the finished banner
if multi expense report and all expenses are finished show the finished banner

**Comment by grgia** [comment]
I think this case 
> if single expense report and all expenses are nonreimbursable show the finished banner
Needs to be covered in [NextStep.php](https://github.com/Expensify/Web-Expensify/blob/1b64677b63f41fd5ef748c5a400285958b9033b1/lib/Report/NextStep.php#L4)

**Comment by grgia** [comment]
https://expensify.slack.com/archives/C03TQ48KC/p1724276550180769 
I noticed next steps are missing the titles here

**Comment by pasyukevich** [comment]
@grgia Is there anything that I can help with?

**Comment by pasyukevich** [comment]
@grgia bump here

**Comment by muttmuure** [comment]
This seems to be done?

**Comment by grgia** [comment]
@muttmuure yep this should be done
